openstudiocore/src/model/Duct_Impl.hpp

```diff
@@ -0,0 +1,107 @@
+/**********************************************************************
+ *  Copyright (c) 2008-2015, Alliance for Sustainable Energy.
+ *  All rights reserved.
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ **********************************************************************/
+
+#ifndef MODEL_DUCT_IMPL_HPP
+#define MODEL_DUCT_IMPL_HPP
+
+#include <model/ModelAPI.hpp>
+#include "StraightComponent_Impl.hpp"
+
+namespace openstudio {
+namespace model {
+
+// TODO: Check the following class names against object getters and setters.
+class Connection;
```

Comment:
Remove any and all Connection declarations or includes from the headers.  Actually you shouldn't need any connection includes they will come from the base class.